deploy/helm/sumologic/values.yaml

```diff
@@ -4125,6 +4141,215 @@ metadata:
+          #   "uid": "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # }
+          # Output Body (JSON): {
+          #   "container_name": "loggercontainer",
+          #   "log": "2001-02-03 04:05:06 loggerlog 1 first line\n",
+          #   "namespace": "default",
+          #   "pod_name": "logger-multiline-4nvg4",
+          #   "run_id": "0",
+          #   "stream": "stdout",
+          #   "uid": "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # }
+          # Input Attributes: _none_
+          # Output Attributes:
+          # - k8s.container.name: "loggercontainer"
+          # - k8s.namespace.name: "default"
+          # - k8s.pod.name: "logger-multiline-4nvg4"
+          # - k8s.pod.uid: "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # - run_id: "0"
+          # - stream: "stdout"
+          - id: copy-attributes
+            type: metadata
+            attributes:
+              stream: 'EXPR($.stream)'
+              k8s.container.name: 'EXPR($.container_name)'
+              k8s.namespace.name: 'EXPR($.namespace)'
+              k8s.pod.name: 'EXPR($.pod_name)'
+              run_id: 'EXPR($.run_id)'
+              k8s.pod.uid: 'EXPR($.uid)'
+
+          # clean-up-log-body takes the values of the `log` field in the JSON Body and puts the value as the sole string value of Body.
+          #
+          # Input Body (JSON): {
+          #   "container_name": "",
+          #   "log": "2001-02-03 04:05:06 loggerlog 1 first line\n",
+          #   "namespace": "default",
+          #   "pod_name": "logger-multiline-4nvg4",
+          #   "run_id": "0",
+          #   "stream": "stdout",
+          #   "uid": "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # }
+          # Output Body (string): "2001-02-03 04:05:06 loggerlog 1 first line\n"
+          #
+          # Input Attributes:
+          # - k8s.container.name: "loggercontainer"
+          # - k8s.namespace.name: "default"
+          # - k8s.pod.name: "logger-multiline-4nvg4"
+          # - k8s.pod.uid: "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # - run_id: "0"
+          # - stream: "stdout"
+          # Output Attributes:
+          # - k8s.container.name: "loggercontainer"
+          # - k8s.namespace.name: "default"
+          # - k8s.pod.name: "logger-multiline-4nvg4"
+          # - k8s.pod.uid: "aed49747-b541-4a07-8663-f7e1febc47d5"
+          # - run_id: "0"
+          # - stream: "stdout"
+          - id: clean-up-log-body
+            type: restructure
+            ops:
+              - move:
+                  from: log
+                  to: $
+    exporters:
+      otlphttp:
+        endpoint: http://${LOGS_METADATA_SVC}.${NAMESPACE}.svc.cluster.local:4318
+  daemonset:
+    ## Set securityContext for containers running in pods in log collector daemonset
+    securityContext:
+    # in order to reliably read logs from mounted node logging paths, we need to run as root
+      fsGroup: 0
+      runAsUser: 0
+      runAsGroup: 0
```

Comment:
I'm not sure if there's any other generic way of guaranteeing we can read the node logs. It seems like K8s distributions have different ideas on what the ACLs should be for these logs. On EKS and microk8s they're world-readable, but on KOPS they're not.